tests/simulator_test.cpp

```diff
@@ -64,25 +65,17 @@ class EnergyCollectorTest : public ::testing::Test {
     return false;
   }
 
-  void
-  printCollectors(const std::vector<std::unique_ptr<objects::EnergyCollector>>
-                      &energyCollectors) const {
+  void printCollectors(const Collectors &energyCollectors) const {
+    int i = 0;
     for (const auto &collector : energyCollectors) {
-      std::cout << "x: " << collector->getOrigin().x()
-                << " y: " << collector->getOrigin().y()
-                << " z: " << collector->getOrigin().z()
-                << " rad: " << collector->getRadius() << "\n";
+      std::cout << i++ << " " << *collector << std::endl;
     }
   }
 
-  float topCollectorZCoordinate(
-      const std::vector<std::unique_ptr<objects::EnergyCollector>>
-          &energyCollectors) {
+  float topCollectorZCoordinate(const Collectors &energyCollectors) {
```

Comment:
Naming. All methods should usually start with a verb.

Not required, but this can be also made shorter like:

> getMaxZ

1. collector is already in the argument
2. coordinate is kind of obvious.